Updating HLT DQM and validation for Hgg paths

mplaner opened the pull request:
This PR adds the offline DQM for HLT as well as the HLT validation for diphoton triggers.

cmsbuild commented:
A new Pull Request was created by  @mplaner  for master.

It involves the following packages:

DQMOffline/Trigger
HLTriggerOffline/Higgs


@cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

dmitrijus commented:
This needs greenlight from @mtosi :)

mtosi commented:
I think it needs the new updates by bhawna
in https://github.com/cms-sw/cmssw/pull/18971

mplaner commented:
@mtosi 
Yeah, I would like to wait until that PR is approved and then merge myself to make sure everything still works correctly for us.

I can do this now, if we're confident that there won't be any further changes to #18971.

mtosi commented:
I think bhawna PR is almost done (just waiting for the integration test)

cmsbuild commented:
Pull request #19023 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

mplaner commented:
I updated and tested after merging PR #18971 into my local area.  I've added the additional plots for subleading photon for doublePhoton triggers.

cmsbuild commented:
Pull request #19023 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

mtosi commented:
@dmitrijus can we start the integration test, please ?

fwyzard commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 20a530917a05fcc68f5374dc1a77f1de77a8ecd4
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/20421/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/20421/summary.html

Comparison Summary:
 - You potentially added 7 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1856393
 - DQMHistoTests: Total failures: 3150
 - DQMHistoTests: Total nulls: 65
 - DQMHistoTests: Total successes: 1853005
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

mtosi commented:
what about 
HLT_Diphoton30EB_18EB_R9Id_OR_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v*
?

mtosi commented on HLTriggerOffline/Higgs/python/hltHiggsValidator_cfi.py:
```diff
@@ -195,20 +195,13 @@
-        ),
+            "HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass90_v",
+            "HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass95_v",
+            "HLT_Diphoton30PV_18PV_R9Id_AND_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v7",
```
are you sure you want to specify v7 ?!?!?!??!?!

cmsbuild commented:
Pull request #19023 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #19023 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

mplaner commented:
Hi Mia,
You were right on both fronts.  I had originally left off the prescaled low-mass paths as I didn't want to clutter the DQM and we don't get much useful information from the DQM for them.  But we want them in the menu, so I've updated with DQM plots.  And I fixed the typo so we won't be restricted to "v7" for the low mass paths in the validation plots.

fwyzard commented:
@cmsbuild please test

cmsbuild commented:
-1

Tested at: b52ee7c45c49e50d9af55e3975789bc13fe48596

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/20499/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>10042.0 step3
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step3_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018.log

</pre>50202.0 step3
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step3_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
Pull request #19023 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

dmitrijus commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 08128fb94538161a5dff25eee4b3fe37ee81acb7
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/20540/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/20540/summary.html

Comparison Summary:
 - You potentially added 12 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1817396
 - DQMHistoTests: Total failures: 49011
 - DQMHistoTests: Total nulls: 130
 - DQMHistoTests: Total successes: 1768089
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

davidlange6 commented:
Your PR is unmergeable. Please have a look and possibly rebase it.

cmsbuild commented:
Pull request #19023 was updated. @ghellwig, @kpedro88, @fabozzi, @Martin-Grunewald, @rekovic, @dmitrijus, @perrotta, @vanbesien, @silviodonato, @cmsbuild, @fwyzard, @GurpreetSinghChahal, @davidlange6, @kkousour, @cerminar, @slava77, @ggovi, @mulhearn, @vazzolini, @kmaeshima, @arunhep, @monttj, @franzoni, @lpernie can you please check and sign again.

perrotta commented:
-1
please rebase (taking in mind that now the master is 93X)

kpedro88 commented:
-1
never merge, only rebase
http://cms-sw.github.io/tutorial-resolve-conflicts.html

cmsbuild commented:
Pull request #19023 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 4c2a866191d29471dcc3de31be6998863bdd069d


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5387f33035f82196d51968058b52bc670f79533c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/NoBPTXMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/HLTTagAndProbeOfflineSource.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/RazorMonitor.cc 
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:1:0:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:109:12: error: redeclaration of 'PhotonME PhotonMonitor::photonEtaME_'
   PhotonME photonEtaME_;
            ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:107:12: note: previous declaration 'PhotonME PhotonMonitor::photonEtaME_'
   PhotonME photonEtaME_;
            ^
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/HLTTagAndProbeOfflineSource.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/RazorMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/DQMGenericTnPClient.cc 
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:1:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:109:12: error: duplicate member 'photonEtaME_'
  PhotonME photonEtaME_;
           ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:107:12: note: previous declaration is here
  PhotonME photonEtaME_;
           ^
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5387f33035f82196d51968058b52bc670f79533c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/git-merge-result

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
Pull request #19023 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

mplaner commented:
Apologies for that error, I had it fixed in my local area, but somehow missed it for the rebase and push.  I've now diffed everything and this commit exactly matches what I've tested.

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 2acd37a67d188ea0079c0ada5373a13c148b7955
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21077/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b1580cdb9b247f8d4a93ce44ec73c8c8213a595e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21077/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21077/git-merge-result

cmsbuild commented:
Comparison job queued.

mtosi commented:
can you please prepare the back-port for 92x ?

mplaner commented:
Hi @mtosi,
For the backport, do I submit a separate PR (or are there additional instructions somewhere)?

Commit:
hgg HLT DQM updates

removed extra include statement

fixed bug

don't book diphoton plots fos single photons

cmsbuild commented:
Pull request #19023 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #19023 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21077/summary.html

Comparison Summary:
 - You potentially added 15 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1755841
 - DQMHistoTests: Total failures: 44278
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1711397
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

fwyzard commented:
@mplaner for backporting there is a tutorial at http://cms-sw.github.io/tutorial-resolve-conflicts.html#backporting-a-pr 

Basically, you create a new branch based on CMSSW 9.2.4, cherry-pick the commits from this branch into it, test, and make a new PR.

cmsbuild commented:
-1

Tested at: 617d587a7b3d7b1494bb155e4adb581385002d18


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/734670c38ffca663bbef9705ca608a007cbd1a80
https://github.com/cms-sw/cmssw/commit/aee2fec0f30749647dda21a12654aa65b73912d7
https://github.com/cms-sw/cmssw/commit/526d3184ff25d7b964ad7c87363c62b50c23783f
https://github.com/cms-sw/cmssw/commit/ccc6917f9cac1f82c933e59268df17f3263c2d65
https://github.com/cms-sw/cmssw/commit/b1580cdb9b247f8d4a93ce44ec73c8c8213a595e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21119/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21119/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21119/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/DQMOffline/Trigger/plugins/BPHMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/DQMOffline/Trigger/plugins/HLTDQMObjSelector.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/DQMOffline/Trigger/plugins/NoBPTXMonitor.cc 
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:1:0:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:121:5: error: narrowing conversion of '((PhotonMonitor*)this)->PhotonMonitor::N_PHI1' from 'unsigned int' to 'int' inside { } [-Werror=narrowing]
     N_PHI1, -MAX_PHI1, MAX_PHI1
     ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:128:5: error: narrowing conversion of '((PhotonMonitor*)this)->PhotonMonitor::N_ETA' from 'unsigned int' to 'int' inside { } [-Werror=narrowing]
     N_ETA, -MAX_ETA, MAX_ETA
     ^
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/CondCore/CondDB/test/testFrontier.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/CondCore/CondDB/test/testRootStreaming.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/CondCore/CondDB/test/testReadWritePayloads.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/CondCore/CondDB/test/testPayloadProxy.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/CondCore/CondDB/test/testConnectionPool.cpp 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:74:33: error: cannot delete expression of type 'std::unique_ptr<GenericTriggerEventFlag>'
  if (num_genTriggerEventFlag_) delete num_genTriggerEventFlag_;
                                ^      ~~~~~~~~~~~~~~~~~~~~~~~~
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-03-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:75:33: error: cannot delete expression of type 'std::unique_ptr<GenericTriggerEventFlag>'
  if (den_genTriggerEventFlag_) delete den_genTriggerEventFlag_;
                                ^      ~~~~~~~~~~~~~~~~~~~~~~~~
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/734670c38ffca663bbef9705ca608a007cbd1a80
https://github.com/cms-sw/cmssw/commit/aee2fec0f30749647dda21a12654aa65b73912d7
https://github.com/cms-sw/cmssw/commit/526d3184ff25d7b964ad7c87363c62b50c23783f
https://github.com/cms-sw/cmssw/commit/ccc6917f9cac1f82c933e59268df17f3263c2d65
https://github.com/cms-sw/cmssw/commit/b1580cdb9b247f8d4a93ce44ec73c8c8213a595e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21119/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21119/git-merge-result

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

fwyzard commented:
@mplaner, can you fix the errors, test locally, and submit an update ?

mplaner commented:
@fwyzard I'll do it, but I'm confused why there were conflicts for this merge.  Locally, everything is up to date and runs without issue (for the nightly build from July 2nd).  I'm traveling today, but should manage to resubmit the PR shortly.

cmsbuild commented:
Pull request #19023 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

mtosi commented:
@mplaner, please, rebase

cmsbuild commented:
Pull request #19023 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

mplaner commented:
Okay, rebased to current master.  Things should run smoothly assuming there aren't other PRs about to get approved which conflict.

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 1863bc843aa215f819f89e838e420184fd58b560


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3ae8d8cefc9af3dd823591d3c6e5b724eb3baffa
https://github.com/cms-sw/cmssw/commit/13803470fe8f47d6cbcc55dfec9b07464e0bef43
https://github.com/cms-sw/cmssw/commit/7ac95672d6a7ef4a1e890ab5c0413bbe4f6a0a86
https://github.com/cms-sw/cmssw/commit/f19409072807f1553ebbe35357dabadf35f91d3e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21174/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21174/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21174/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>10042.0 step3
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step3_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>50202.0 step3
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step3_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018.log

</pre>25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3ae8d8cefc9af3dd823591d3c6e5b724eb3baffa
https://github.com/cms-sw/cmssw/commit/13803470fe8f47d6cbcc55dfec9b07464e0bef43
https://github.com/cms-sw/cmssw/commit/7ac95672d6a7ef4a1e890ab5c0413bbe4f6a0a86
https://github.com/cms-sw/cmssw/commit/f19409072807f1553ebbe35357dabadf35f91d3e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21174/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21174/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
Pull request #19023 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

mtosi commented:
@cmsbuild, please test

> On 5 Jul 2017, at 17:37, cmsbuild <notifications@github.com> wrote:
> 
> Pull request #19023 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 3ea981e7058fc90b5f04e6c2f73f7fde3c2efd93
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21196/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3ae8d8cefc9af3dd823591d3c6e5b724eb3baffa
https://github.com/cms-sw/cmssw/commit/13803470fe8f47d6cbcc55dfec9b07464e0bef43
https://github.com/cms-sw/cmssw/commit/7ac95672d6a7ef4a1e890ab5c0413bbe4f6a0a86
https://github.com/cms-sw/cmssw/commit/f19409072807f1553ebbe35357dabadf35f91d3e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21196/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21196/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21196/summary.html

Comparison Summary:
 - You potentially added 5 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1786696
 - DQMHistoTests: Total failures: 17980
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1768550
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

dmitrijus commented:
The description says "This PR supersedes #18971", but that PR is already merged. So, this PR one cannot supersede that one.
Please update the description.

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

mtosi commented:
can you point me to the back-port in 92x, please ?
thanks !

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -167,6 +179,36 @@ void PhotonMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
   bookME(ibooker,photonEtaPhiME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin, eta_binning_.xmax,phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
   setTitle(photonEtaPhiME_,"#eta","#phi"); 
 
+  //for diphotons
```
hi @mplaner - if nphotons=1 these don't get used. Please don't fill them in that case.

davidlange6 replied:
fill -> book

mplaner replied:
Thanks, that's a good point and has been updated.

cmsbuild commented:
Pull request #19023 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

mplaner commented:
@mtosi The 92X version is here: #19612

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
-1

Tested at: cd642f667b243b6534e6c5d3142f42d7f824f34c


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/06199984952d9a6468923f7f1e47bed589272363
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21291/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21291/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21291/summary.html

I found follow errors while testing this PR

**Failed tests:  AddOn**

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun_MC.root : FAILED - time: date Mon Jul 10 11:06:38 2017-date Mon Jul 10 11:05:26 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 11:06:38 2017-date Mon Jul 10 11:05:26 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 11:06:38 2017-date Mon Jul 10 11:05:26 2017 s - exit: 21248
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_25ns --magField 38T_PostLS1 : FAILED - time: date Mon Jul 10 11:06:06 2017-date Mon Jul 10 11:05:28 2017 s - exit: 21248
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_PRef_MC.root : FAILED - time: date Mon Jul 10 11:06:22 2017-date Mon Jul 10 11:05:30 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 11:06:22 2017-date Mon Jul 10 11:05:30 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 11:06:22 2017-date Mon Jul 10 11:05:30 2017 s - exit: 21248
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_2016 --magField 38T_PostLS1 : FAILED - time: date Mon Jul 10 11:06:45 2017-date Mon Jul 10 11:05:55 2017 s - exit: 21248
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun2016 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun2016_MC.root : FAILED - time: date Mon Jul 10 11:07:59 2017-date Mon Jul 10 11:06:08 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun2016.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 11:07:59 2017-date Mon Jul 10 11:06:08 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun2016,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun2016 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun2016_MC.root --fileout file:RelVal_Raw_GRun2016_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 11:07:59 2017-date Mon Jul 10 11:06:08 2017 s - exit: 21248
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016,Run2_HI --magField 38T_PostLS1 --fileout file:RelVal_Raw_HIon_MC.root : FAILED - time: date Mon Jul 10 11:08:29 2017-date Mon Jul 10 11:06:46 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 11:08:29 2017-date Mon Jul 10 11:06:46 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016,Run2_HI --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_HIon_MC.root --fileout file:RelVal_Raw_HIon_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 11:08:29 2017-date Mon Jul 10 11:06:46 2017 s - exit: 21248
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_PIon_MC.root : FAILED - time: date Mon Jul 10 11:14:39 2017-date Mon Jul 10 11:12:01 2017 s - exit: 21248
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02480/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_3_X_2017-07-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Jul 10 11:14:39 2017-date Mon Jul 10 11:12:01 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_MC.root --fileout file:RelVal_Raw_PIon_MC_HLT_RECO.root : FAILED - time: date Mon Jul 10 11:14:39 2017-date Mon Jul 10 11:12:01 2017 s - exit: 21248



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/06199984952d9a6468923f7f1e47bed589272363
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21291/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21291/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21291/summary.html

Comparison Summary:
 - You potentially added 3 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1884112
 - DQMHistoTests: Total failures: 14541
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1869405
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

fwyzard commented:
The failing tests are unrelated to this PR

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

fwyzard commented:
Please merge #19684 which includes this and fixes conflicts.

cmsbuild commented:
+1
Tested at: cd642f667b243b6534e6c5d3142f42d7f824f34c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21364/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/92ee79880f3dc245397a00e6acced67d57d310cc
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21364/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21364/git-merge-result

cmsbuild commented:
Comparison job queued.

davidlange6 commented:
merge

fwyzard commented:
David,
may I very politely ask "what the fuck ?" ?

I spent some time to go over all the HLT DQM PRs that were OK from the implementation point of view, merge all of them taking care of the conflicts, and write an explicit message on GitHub in each of them.

And the first thing you do is merge one of them at random messing up the confluct resolution I have already done.

.Andrea

davidlange6 commented:
think of it as me being totally lost - I guess you propose to close all of the prs, make me redo the review on something too big to review? To be honest, the opposite approach of small one-feature-one-PR pull requests would be better (but perhaps impossible for reasons I've not understood yet)


> On Jul 11, 2017, at 11:47 AM, Andrea Bocci <notifications@github.com> wrote:
> 
> David,
> may I very politely ask "what the fuck ?" ?
> 
> I spent some time to go over all the HLT DQM PRs that were OK from the implementation point of view, merge all of them taking care of the conflicts, and write an explicit message on GitHub in each of them.
> 
> And the first thing you do is merge one of them at random messing up the confluct resolution I have already done.
> 
> .Andrea
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

fwyzard commented:
No, I propose something much simpler:
  - we leave the individual PRs open for the moment
  - I merge them one at a time in a single PR, leaving the individual commits untouched, adding only the resolution of the merge conflicts and minor fixes
  - we do not re-review the "merged PR", since all the individual changes were already reviewd
  - once the tests pass, you merge the "merged PR"
  - this should automatically set the original PRs as "merged" as well

If you agree with the plan, I will re-do the "merged PR" based on the current 9.3.X.

fwyzard commented:
The opposite approach (one PR per feature) is what we have been doing so far.
It works well for reviewing the changes, but is a mess for merging PRs that need to modify the same files, as you have seen.

davidlange6 commented:
> On Jul 11, 2017, at 11:58 AM, Andrea Bocci <notifications@github.com> wrote:
> 
> No, I propose something much simpler:
> 
> 	• we leave the individual PRs open for the moment
> 	• I merge them one at a time in a single PR, leaving the individual commits untouched, adding only the resolution of the merge conflicts and minor fixes
> 	• we do not re-review the "merged PR", since all the individual changes were already reviewd
> 	• once the tests pass, you merge the "merged PR"
> 	• this should automatically set the original PRs as "merged" as well
> If you agree with the plan, I will re-do the "merged PR" based on the current 9.3.X.

ok - fine, but then don't merge small PRs into the merged PR until they are reviewed.

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21364/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2012093
 - DQMHistoTests: Total failures: 117
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2011810
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files
